packages/nimble-components/src/rich-text-editor/specs/README.md

```diff
@@ -128,7 +131,22 @@ application's performance is enhanced as the operation is performed only once, t
 
 _Methods_
 
--   none
+-   `hideFooter()` - hides the footer section that contains all formatting options and the `footer-actions` slot.
+-   `clearContent()` - clears the content in the editor through [clearContent()](https://tiptap.dev/api/commands/clear-content) command in Tiptap.
+
+_Open Discussion_
+
+-   We are proposing a design that involves displaying only placeholder text initially and footer section will be hidden. Upon focusing the editor, we will capture
+    Tiptap's [focus event](https://tiptap.dev/api/events#focus) to show the footer section. As per the
+    UX for comments feature, we will revert back to the original state (i.e,. without the footer section) only if the user clicks the `cancel` or `ok` button. Therefore,
+    we came to this decision of exposing the above `hideFooter()` and `clearContent()` methods.
+-   There isn't a specific reason to expose two separate methods when considering the comments feature; they could be combined into a single method. However, we are uncertain about the
+    appropriate name for this method. One idea we have is to name it `resetEditor()`, combining the functionality of hiding the footer and clearing the existing content in the editor. This method
+    would not modify the markdown value and so we are not sure `resetEditor()` is the appropriate name for it. So, the question is, can we consolidate both functionalities into a single method
+    and find a suitable name for it?
+-   The rationale behind not suggesting it as a property/attribute is that we don't see a compelling reason to expose both showing and hiding functionalities of the footer section to the client.
+    Instead, our proposal is to only expose a method that allows the client to hide the footer when necessary. The rich text editor can show the footer section exclusively when focused, while
```

Comment:
I guess the part I'm missing is what is the operation the application would do to show the footer in the first place? The API is unbalanced if you have a 'hide' but not a 'show'. A 'displayFooter' boolean attribute (or 'hideFooter' if we want the default value to be 'false') would work for both cases

Reply:
I think writing down the following would help us come to a good API:
1. a list of the states that the component can be in
2. a list of the operations that can trigger a transition between those states
3. for each operation, a proposal of whether it can be detected by the rich-text-editor and if so whether the editor should automatically transition. Or or whether the client needs an API to tell the editor to transition.

Once we understand those states and transitions we can propose either attributes that represent the states or methods that represent the transitions (or both).

For example, I believe the states are something like this: (focused and not focused might be additional states)
1. no text, not showing footer
2. no text, showing footer
3. has text, showing footer

1 is the initial state

1->2 is triggered when the user focuses the editor. The editor can detect this so the rich text component could make this transition automatically. Should it?

2->1 requires clicking the cancel button which is provided by the client so the rich text editor needs to provide an API to make that transition

Reply:
> 2->1 requires clicking the cancel button which is provided by the client so the rich text editor needs to provide an API to make that transition

For slotted button cancel operation, we definitely need an attribute `footer-hidden` (API) to allow clients to set the footer state for those button. Making an attribute API also makes it symmetric to show or hide the corresponding UI part. 

> 1->2 is triggered when the user focuses the editor. The editor can detect this so the rich text component could make this transition automatically. Should it?

For comments workflow, this transition applies only when adding a new comment through `sl-new-comment` component which always has the rich text editor available at top of the page. For editing/deleting a comment we quickly replace the editor with viewer on cancel / OK. 

For other use cases like description boxes as well this might likely apply based on the requirements. 

If it's too early for us to decide on a default footer state (hide initially and show in focus) for different use cases of rich text editor, can we defer this decision and enable clients to decide the visibility for their use cases by prioritizing implementation of `footer-hidden` API? Because this will allow clients to set this state based on their requirements.

Reply:
Just so the decision is spelled out, is the idea that the rich-text-editor itself won't define the UX of only showing the toolbar when it has focus and that is an application specific / usage specific workflow? Does that sound like the decision we are aligning on? @jattasNI @rbell517 

We can imagine that there will be enough variations of usage of rich text controls that we don't want to constrain / make consistent that part of the UX?

I can see that being a reasonable starting decision and the proliferation of usage of the rich-text-control is expected to be minimal, so changing the behavior may be reasonable in the future (for instance blazor apps desire the same UX for non-comment usages, maybe general description editing or something).

If that's the current decision then it should be clarified in the spec that the behavior described around footer hiding interactions is how an example application will use the `footer-hidden` property and not that the behavior of the rich text component itself:

https://github.com/ni/nimble/blob/91cb5fff3baf50cc76fb82248d47d1985b3b82ae/packages/nimble-components/src/rich-text-editor/specs/README.md?plain=1#L159-L161

Reply:
I have updated the behavior of the footer hiding interactions and how the example application can utilize this API.